### PR TITLE
Fix PHP modules not re-compiling on PHP version changes

### PR DESCRIPTION
**Pull request by oxyc**
Would be nicer if this could be done upstream in the php extension roles though... Now stuff such as solr is rebuilt as well.

### CODE REVIEW

**Comment by geerlingguy** [comment]
Hmm... I'd be willing to do things in the `php` role, but it's a lot more effort to try to change _all_ the `php-*` roles.

**Comment by geerlingguy** [comment]
Especially since all the annoying changes are for Debian only (RHEL repos use the same package names, paths, etc. and unless you use Software Collections you just install one version at a time).

**Comment by oxyc** [comment]
This one affects RHEL as well. The extension never recompiles even if the PHP version changes. It's just moved back into place.

**Comment by oxyc** [comment]
Also, it's pretty rare that people switch PHP versions. Re-building solr/daemonize/etc isnt ideal, but it barely increases the build time anyways..

**Comment by oxyc** [comment]
Nevermind, they are re-compiled, just not moved into place :) @joestewart https://github.com/geerlingguy/drupal-vm/issues/1076#issuecomment-268988244

**Comment by geerlingguy** [comment]
This would still be needed after the upstream fix in the Xdebug role, correct? And would we need to fix anything in the XHProf role too?

**Comment by oxyc** [comment]
@geerlingguy actually this is the only PR that is required for the fix. This is the solution suggested by @thom8 in https://github.com/geerlingguy/ansible-role-php-xdebug/pull/41#issuecomment-269059900.

Unfortunately I don't have time to test it and I'll be offline until Monday. 

In addition to this @joestewart (https://github.com/geerlingguy/drupal-vm/issues/1076#issuecomment-268988244) mentioned that the XDebug extension doesn't upgrade when you change the version (the extension version), this is a separate issue though. https://github.com/geerlingguy/ansible-role-php-xdebug/pull/41 has his fix rebased but bundled with some other commits that should not be merged if we go with assigning the `workspace` here in the playbook instead.

File: provisioning/tasks/init-debian.yml [hunk 1]
```diff
@@ -55,15 +55,6 @@
...
 
-- name: Purge PHP modules directory.
-  file:
-    path: "{{ item }}"
-    state: absent
-  when: php_purge.changed
-  with_items:
-    - "/usr/lib/php5/modules"
-    - "/usr/lib/php/modules"
-
```
Comment (oxyc) [inline provisioning/tasks/init-debian.yml]: Actually, I think we need to test that memcached/redis extensions still work on version change without this. Unfortunately I wont have time to do that until monday.

**Comment by geerlingguy** [comment]
@oxyc - I merged https://github.com/geerlingguy/ansible-role-php-xdebug/pull/42, so it might be easier to just pull in the updated version of that role (2.3.0).

**Comment by geerlingguy** [comment]
Ah, true. Post-holiday confusion from having like 40 new issues across my roles :D

**Comment by geerlingguy** [comment]
(Apparently everyone but me works on my roles over Christmas.)

**Pull request merged**